Fix provider ref names (#34016)
dotnet/docs: docs/orleans/streaming/streams-quick-start.md

> ## Producing events

Reviewer: ```suggestion
## Produce events
```

<!-- markdownlint-disable MD044 -->
:::zone target="docs" pivot="orleans-7-0"
<!-- markdownlint-enable MD044 -->

> Producing events for streams is relatively easy. You should first get access to the stream provider which you defined in the config above (`"StreamProvider"`) and then choose a stream and push data to it.

Reviewer: ```suggestion
It's relatively easy to produce events for streams. You should first get access to the stream provider that you defined in the config previously (`"StreamProvider"`), and then choose a stream and push data to it.
```

:::zone target="docs" pivot="orleans-3-x"
<!-- markdownlint-enable MD044 -->

> Producing events for streams is relatively easy. You should first get access to the stream provider which you defined in the config above (`"SMSProvider"`) and then choose a stream and push data to it.

Reviewer: ```suggestion
It's relatively easy to produce events for streams. You should first get access to the stream provider that you defined in the config previously (`"SMSProvider"`), and then choose a stream and push data to it.
```

> ## Subscribing and receiving streaming data

Reviewer: ```suggestion
## Subscribe to and receive streaming data
```

> For receiving data, we can use implicit/explicit subscriptions, which are fully described in other pages of the manual. Here we use implicit subscriptions, which are easier. When a grain type wants to implicitly subscribe to a stream, it uses the attribute [[ImplicitStreamSubscription(namespace)]](xref:Orleans.ImplicitStreamSubscriptionAttribute).

Reviewer: ```suggestion
For receiving data, you can use implicit/explicit subscriptions, which are fully described in other pages of the docs. This example uses implicit subscriptions, which are easier. When a grain type wants to implicitly subscribe to a stream, it uses the attribute [[ImplicitStreamSubscription(namespace)]](xref:Orleans.ImplicitStreamSubscriptionAttribute).
```

Reviewer 2: How about "For receiving data, use..." - just curious if we are keen to avoid direct references to the reader? (yes, this could be chalked up as personal preference)

Reviewer: Either way is probably fine (just not "we"). The style guide says to have a "conversational tone". Also see https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=25524.

> For our case we'll define a `ReceiverGrain` like this:

Reviewer: ```suggestion
For your case, define a `ReceiverGrain` like this:
```

Reviewer: Although it looks like other text e.g. lines 129-131 also uses "we".
